core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/handler/AwsAsyncClientHandler.java

```diff
@@ -61,7 +58,8 @@ public <InputT extends SdkRequest, OutputT extends SdkResponse, ReturnT> Complet
 
     @Override
     protected <InputT extends SdkRequest, OutputT extends SdkResponse> ExecutionContext
-        invokeInterceptorsAndCreateExecutionContext(ClientExecutionParams<InputT, OutputT> executionParams) {
+        invokeInterceptorsAndCreateExecutionContext(ClientExecutionParams<InputT, OutputT> executionParams,
+                                                    SdkClientConfiguration clientConfiguration) {
```

Comment:
This is a protected API. Changing it has the potential to break old clients. Is that a risk here?

Reply:
We have an override in the parent class, so this should be backwards compatible.

Reply:
Nice! If we put `clientConfiguration` in the `ClientExecutionParams` can we avoid needing to add another method here? It's not clear to me right now why this is a separate parameter instead of part of the client execution params.